src/components/formio/select.tsx

```diff
@@ -202,10 +211,16 @@ function Select<
           menuPlacement="auto"
           {...field}
           {...props}
-          onChange={newValue => {
+          onChange={(newValue, action) => {
             const isSingle = !Array.isArray(newValue);
-            const normalized = isSingle ? [newValue] : newValue;
-            const rawValues = normalized.map(val => val?.[valueProperty] ?? null);
+
+            // for multiple, the value is reset to empty array (`[]`)
+            if (action.action === 'clear' && isSingle) {
+              newValue = emptyValue;
+            }
+
+            const normalized = (isSingle ? [newValue] : newValue) as MultiValue<Option>;
```

Comment:
Why do we explicitly declare the result type here?

Reply:
otherwise the typescript compiler complains about implicit `any` types :)